### ISSUE TITLE
Allow user to retrieve the N last commits

### ISSUE THREAD

**Issue by francescmm** [francescmm/GitQlient#163]
Instead of retrieving all the commits of the repo it would be nice to have be able to configure how many commits GitQlient gets.

This improves performance and reduces the memory for very big repos.

**Comment by francescmm** [comment]
This should help you with the performance at least a bit, @band-a-prend

**Comment by band-a-prend** [comment]
@francescmm, thank you!

P.S.
I recently [packaged GitQlient for gentoo](https://gitweb.gentoo.org/repo/proj/guru.git/tree/dev-vcs/gitqlient), currently not in the main tree but in the [GURU - gentoo user repository](https://wiki.gentoo.org/wiki/User:RarogCmex/Project:GURU/End_User_Guide) repository.
There are added `gitqlient-1.2.0` and `gitqlient-9999` to build directly from `develop` (not `master`) branch.

**Comment by francescmm** [comment]
That's awesome, thanks!

Just let you know that I use develop as _in progress_ branch. So I'd say it's unstable, but if that's okay for you/gentoo, it is for me :+1:

**Comment by band-a-prend** [comment]
Yes, `live`, i.e `-9999`, gentoo build scripts are assumed as the latest upstream development sourcecode tarball at the moment of build process. Therefore it never has specific `arch` and always unstable.

**Comment by band-a-prend** [comment]
@francescmm, I recently (~ a week ago) noted the disappearing of `develop` branch for a while and then it appeared again. Some distributives, e.g. Arch and Gentoo, could package "bleeding edge" of upstream code in their official or user repositories. Could I consider the `future-1.4.0` branch as such "edge" to try new unstable features? As I see the `develop` currently looks like inactive.

**Comment by francescmm** [comment]
@band-a-prend don't worry about the develop disappearing again! What happens is that I have activated the "remove branch after PR is merge". What I didn't expect is that if I have a PR open and I manually merge the branch into the other the PR closes automatically and the branch goes away. I've disable it.

The plan is to have _**develop**_ as unstable but compiling branch, _**master**_ as the official releases branch and then the others where I do experiments.

For the distros I'd follow the **_develop_** branch. The future-1.4.0 was meant to be some experiments but I finished most of them way faster than expected, so I'm gonna merge them now in develop. :smile:
